phpcs/QualityAssurance/Sniffs/Generic/CredentialsSniff.php

```diff
@@ -46,15 +46,15 @@ public function register()
      * @param int  $stackPtr  The position of the current token
      *                        in the stack passed in $tokens.
      *
-     * @return void
+     * @return int
      */
     public function process(File $phpcsFile, $stackPtr)
     {
         $end = (count($phpcsFile->getTokens()) + 1);
         $filePath = $phpcsFile->getFilename();
         $fileName = basename($filePath);
 
-        if (preg_match('/docker-compose*/', strtolower($fileName)) === false) {
+        if (preg_match('/docker-compose*/', strtolower($fileName)) !== 1) {
```

Comment:
Couldn't we use the `i` modifier here to avoid `strtolower()`?

Reply:
Indeed is kind of micro-optimisation.

The str_contains would be the ideal, but we still need to support the 7.4 in the 8.x branch (for now)
In terms of performance and considering the used pattern, it seems to be the `strpos` the best option, not using the strtolower, but making a comparison with double equal.

Have a look to this test https://3v4l.org/tbdAg

Also realize that adding the `^` to the pattern makes it much faster.

Also noticed some issue with the variables being checked, one of them is wrong and is missing the nextcloud ones, will need to fix them.

> Not sure what you mean by that though.

I just meant that we usually use the `strtolower()` when need to compare the filename, like in the 
phpcs/QualityAssurance/Sniffs/InfoFiles/RequiredSniff.php#L54
phpcs/QualityAssurance/Sniffs/InstallFiles/FunctionDeclarationsSniff.php#L52
phpcs/QualityAssurance/Sniffs/InstallFiles/HookUpdate0Sniff.php#L52
phpcs/QualityAssurance/Sniffs/InstallFiles/HookUpdateNSniff.php#L52
phpcs/QualityAssurance/Sniffs/InstallFiles/InstallUpdateCallbacksSniff.php#L52

Reply:
Unfortunately the phpcs will not let me use a double equal.
`Operator != prohibited; use !== instead`